Dockerfile

```diff
@@ -9,6 +9,7 @@ COPY . ./
 RUN mv .env.example .env
 
 # Generate Prisma client
+RUN npm install
```

Comment:
Zum Zweck der Dokumnetation hole ich mal etwas weiter aus: 

Nachteile: 
- Mehrfaches ausführen von npm install in der CI pipeline

Vorteile:
- Docker Image kann auch ohne lokal installiertes node gebaut werden
- Dockerfile ist in sich "abgeschlossen"

Kurz zu den Vorteilen: 
Ich glaube kaum dass die Container von nicht-Entwicklern gebaut werden, was heißt man kann annehmen das node installiert ist. 

Es gibt vermutlich zwei konkurrierende Ansätze für das bauen unserer Anwendung hier. 
1. Bauen vollständig durch docker. Die Idee ist, man kann unsere Anwendung klonen und ein `docker build` reicht auch aus um eine Anwendung zu erhalten, mehr ist nicht notwendig. 
2. Das erstellen eines Images ist nur ein Schritt in einem größeren Build-Prozess. (`npm i` -> `npm build` -> ... -> `docker build`)

Ich tendiere aus verschiedenen Gründen zur zweiten Option.
Wir werden ohnehin ein mehrschritten Bauprozess manuell via node für Entwickler untersützen müssen (den werden wir nicht los).  Docker könnte dann nach und nach nur der letzte Step, das packetieren und ausliefern der Anwendung werden. 

TLDR: 
Vielleicht wäre folgendes ausreichend?

```suggestion
ADD node_modules /app/node_modules
```

Reply:
Insofern wir den Build-Prozess nicht weiter anpassen (Richtung ein fertiges Image gehen), finde ich einen atomaren Build (der sich nicht auf das Host-Environment verlässt; bzw Fall 1) prinzipiell sauberer.

Wäre es nicht eine Option das zukünftig nur im master so nach 1 zu bauen?

Reply:
> In dem Fall könnte vermutlich aber auch die nachfolgende Zeile gelöscht werden: npx prisma generate. Wenn ich es richtig im Kopf habe, baut es den Client im node_modules Ordner.

Ah klasse. Das hatte ich so gar nicht im Kopf.

> Insofern wir den Build-Prozess nicht weiter anpassen (Richtung ein fertiges Image gehen), finde ich einen atomaren Build (der sich nicht auf das Host-Environment verlässt; bzw Fall 1) prinzipiell sauberer.

Ja da bin ich bei dir. Deshalb wird innerhalb der CI Pipeline auch eine eigene Umgebung mit Docker gebaut. Die einzelnen Schritte die dort durchgeführt werden, sind aber näher an den Prozessen dran, die auch entwickler durch, außerhalb eines Docker Builds.

> Wäre es nicht eine Option das zukünftig nur im master so nach 1 zu bauen?

Ich weiß nicht ob ich dich richtig verstanden habe. Ich würde ungerne den Buildprozess den Entwickler durchführen müssen, vom Branch abhängig machen. Die Schritte innerhalb einer Dockerfile zu isolieren funktioniert auch nicht vollumfänglich, da wir ja tests mit einer Datenbank laufen lassen müssen. Ich vermute du meintest die CI/CD Seite?

Die Idee die du angesprochen hast, versuchen wir auf Entwickler-Setie ja mit den dev-environments umzusetzen, sodass jeder Entwickler die gleiche Umgebung erhält, egal auf welchem System er ist. 
Die CI-Pipeline hat eine abgespeckte, aber nachgestellte Environemtn (hier könnte man auch überlegen ob man zukünftig auf den lightweight-weg in der Pipeline verzeichtet, und auch einfach die Dev-Env die auch die Entwickler verwenden, nutzt um die Build maximal reproduzierbar zu machen).